src/ai/backend/manager/services/artifact_registry/service.py

```diff
@@ -219,9 +220,18 @@ async def get_registry_meta(
         self, action: GetArtifactRegistryMetaAction
     ) -> GetArtifactRegistryMetaActionResult:
         log.info("Getting artifact registry meta with id: {}", action.registry_id)
-        registry_meta = await self._artifact_registry_repository.get_artifact_registry_data(
-            action.registry_id
-        )
+        if action.registry_id:
+            registry_meta = await self._artifact_registry_repository.get_artifact_registry_data(
+                action.registry_id
+            )
+        elif action.registry_name:
+            registry_meta = (
+                await self._artifact_registry_repository.get_artifact_registry_data_by_name(
+                    action.registry_name
+                )
+            )
+        else:
+            raise InvalidAPIParameters("Either registry_id or registry_name must be provided.")
```

Comment:
I don't like this approach. but this time I'll merge it. We should refactor this identifier behaviour.